esm2quinox/_esm2.py

```diff
@@ -119,12 +119,14 @@ def tokenise(
 
 class ESM2Result(eqx.Module):
     """The output result from calling `esm2quinox.ESM2.__call__`. Has the `.hidden`
-    representation from the final layer of the model, and the `.logits` (pre-softmax)
-    from mapping that hidden layers through a prediction head.
+    representation from the final layer of the model, the `.logits` (pre-softmax)
+    from mapping that hidden layers through a prediction head, and `.all_hidden`
+    which contains the outputs from all layers (including the final layer).
     """
 
     hidden: Float[Array, "length embed_size"]
     logits: Float[Array, "length alphabet_size"]
+    all_hidden: Float[Array, "num_layers length embed_size"]
```

Comment:
Thanks for the contribution!

Looking at this line, I think it would be more efficient to represent the hidden state as a `list[Float[Array, "length embed_size"]]`.

The typical use-case in which a user needs only a few of the layers would then make it possible for the compiler to DCE the remaining elements of the list.